tangelo/algorithms/classical/mp2_solver.py

```diff
@@ -132,12 +147,184 @@ def get_mp2_amplitudes(self):
+                   set(self.molecule.frozen_virtual[0]) != set(self.molecule.frozen_virtual)):
+                    raise ValueError(f"Only identical frozen orbitals for alpha and beta are supported in {self.__class__.__name__}")
+            focc = np.array(self.molecule.frozen_occupied)
+            fvir = np.array(self.molecule.frozen_virtual)
+            if np.any(focc > n_frozen_occ-1) or np.any(fvir < self.molecule.n_mos-n_frozen_vir):
+                raise ValueError(f"{self.__class__.__name__} does not support freezing interior orbitals")
+
+        if not self.molecule.uhf:
+            ref = 'rhf' if self.molecule.spin == 0 else 'rohf'
+        else:
+            ref = 'uhf'
+        self.backend.set_options({'basis': self.basis, 'mcscf_maxiter': 300, 'mcscf_diis_start': 20,
+                                  'opdm': True, 'tpdm': True, 'frozen_docc': [n_frozen_occ], 'frozen_uocc': [n_frozen_vir],
+                                  'reference': ref})
+
+        energy, self.mp2wfn = self.backend.energy('mp2', molecule=self.molecule.solver.mol,
+                                                  basis=self.basis, return_wfn=True)
+        return energy
+
+    def get_rdm(self):
+        """Calculate the 1- and 2-particle reduced density matrices.
+
+        Obtaining MP2 rdms from Psi4 is not supported in Tangelo.
+
+        Raises:
+            RuntimeError: Not implemented at this time"""
+        raise RuntimeError("Returning MP2 rdms from Psi4 is not currently supported in Tangelo")
+
+    def get_mp2_amplitudes(self):
+        """Compute the double amplitudes from the MP2 perturbative method, and
+        then reorder the elements into a dense list. The single (T1) amplitudes
+        are set to a small non-zero value. The ordering is single, double
+        (diagonal), double (non-diagonal).
+
+        Returning MP2 amplitudes from Psi4 is not supported in Tangelo
+
+        Raises:
+            NotImplementedError: Not implemented at this time"""
+        raise RuntimeError("Returning MP2 amplitudes from Psi4 is not currently supported in Tangelo")
+
+
+available_mp2_solvers = {'pyscf': MP2SolverPySCF, 'psi4': MP2SolverPsi4}
+
+
+def get_mp2_solver(molecule: SecondQuantizedMolecule, solver: Union[None, str, Type[ElectronicStructureSolver]] = default_mp2_solver, **solver_kwargs):
+    """Return requested target MP2SolverName object.
+
+    Args:
+        molecule (SecondQuantizedMolecule) : Molecule
+        solver (string or Type[ElectronicStructureSolver] or None): Supported string identifiers can be found in
+            available_mp2_solvers (from tangelo.algorithms.classical.mp2_solver). Can also provide a user-defined MP2 implementation
+            (child to ElectronicStructureSolver class)
+        solver_kwargs: Other arguments that could be passed to a target. Examples are solver type (e.g. mcscf, mp2), Convergence options etc.
+     """
```

Comment:
For this guy, let's document the exceptions, as it currently can return 3 types.